### ISSUE TITLE
[BUGS] the plugin is making cracking sound no matter if i use VST3, LV2 or Clap

### ISSUE THREAD

**Issue by nikora420** [midilab/jc303#21]
i download the plugin from the site i use the VST3, LV2 and Claps and both of them are making cracking sound on Reaper

here a recording of the cracking sound
https://github.com/user-attachments/assets/016eeb1f-80af-4fa9-9b18-6d461261bf36

OS: Ubuntu 24.04
DAW: Reaper 7.22

**Comment by midilab** [comment]
hi @nikora420 ,

Since i don't have Reaper i can't reproduce the same scenario.

Can you please tell me the following for your scenario:

a) sampler rate
b) buffer size
c) can you test other synth plugin with same conditions(sampler rate and buffer size) as the one you mentioning?

**Comment by nikora420** [comment]
> hi @nikora420 ,
> 
> Since i don't have Reaper i can't reproduce the same scenario.
> 
> Can you please tell me the following for your scenario:
> 
> a) sampler rate b) buffer size c) can you test other synth plugin with same conditions(sampler rate and buffer size) as the one you mentioning?
a. sampler rate: 48kHz
b. buffer size is 1024
c. other then a forked of vital crashing the other synth plugin work fine with the same sampler rate and buffer size

**Comment by midilab** [comment]
thanks for the info. 

can you try with lower volume? looks like the volume is cliping.

other than that i see no other problem, the last test after the volume could be a local compile using github source code on your machine.

**Comment by nikora420** [comment]
i try it lower volume still the same

**Comment by midilab** [comment]
do you have any other DAW where you can test other than Reaper?

**Comment by c3ntr0n800** [comment]
Another REAPER user here, I am also experiencing crackling with the recent release. It's very noticeable when moving the knobs, I ended up going to the previous version, since quite a few of my projects uses modulation on the VST

**Comment by midilab** [comment]
@c3ntr0n800 thanks for the reporting.

is anyone here able to compile the latest github version on your own machine for a test?

**Comment by midilab** [comment]
Another test is the same source code compiled using the old skin.

@c3ntr0n800 and @nikora420 can you guys please test this vst3 version against Reaper for me?

If everything works fine with this attached version, it will point to the new skin implementation as the source of the problem.

[JC303-test-old-skin-vst3.zip](https://github.com/user-attachments/files/16906077/JC303-test-old-skin-vst3.zip)

**Comment by c3ntr0n800** [comment]
Oh I see this is a build for Linux. Unfortunately, my other PCs with Linux are unavailable at the moment. Perhaps I can install a VM to test this later

**Comment by c3ntr0n800** [comment]
Apologies for going quiet for 3 days, I was unable to get around to testing the Linux versions via VM until tonight... But now I can confirm that the crackling occurs when modulating the VST, both with the latest release and your test build

**Comment by midilab** [comment]
> Apologies for going quiet for a few days, I was unable to get around to testing the Linux versions via VM until tonight... But now I can confirm that the crackling occurs when modulating the VST, both with the latest release and your test build

Thanks for the tests, but testing inside a VM is not ideal scenario since there is a lots of processing layers between the VM and you main OS that will impact on any realtime operation badlly.

I can sent you other format for test on a native OS instead of a VM.

Can you inform your main OS, DAW and computer processor model? 

I've download reaper to test, but i couldn't reproduce the bahaviour of cracking sounds neither on modulating parameters(@c3ntr0n800) or playing from a keyboard(@nikora420)

In that case things becomes really hard to debug because i cant reproduce it. Said that, the only way out could be doing tests on the machines of those reporting the cracking so we can try to figure out what is the root of the issue.

**Comment by c3ntr0n800** [comment]
Ah, so I guess we're the unlucky ones. lol

Anyways, my main OS is Windows 10, DAW is REAPER 7.22, and CPU is Intel Core i5-8500

**Comment by midilab** [comment]
Can you try this VST3 windows version to test if the crack sound persist on Reaper please?

[jc303-cracksound-test.zip](https://github.com/user-attachments/files/16961988/jc303-cracksound-test.zip)

**Comment by c3ntr0n800** [comment]
Oh nice, this build fixed the crackling modulation on my end. What was the change made with this?

**Comment by midilab** [comment]
Great! thanks for the tests so far, i will soon release the binaries with modifications:

> What was the change made with this?

https://github.com/midilab/jc303/commit/6fc925db3bebcbb901adb55613ede6ea8f152cb8

**Comment by midilab** [comment]
@c3ntr0n800 Could you please test the last fix release to check if everything is still ok about the issue?

https://github.com/midilab/jc303/releases/tag/v0.11.1

**Comment by c3ntr0n800** [comment]
Yep, it's all good for me now! Hope it's all good for the guy who opened the issue too

**Comment by midilab** [comment]
Thank you for the test! i hope you have some good acid time with jc303.
